hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -896,6 +919,15 @@ def stop_example(self, discard=False):
         self.depth -= 1
         assert self.depth >= -1
         self.__example_record.stop_example(discard)
+
+        labels_for_structure = self.labels_for_structure_stack.pop()
+
+        if not discard:
+            if self.labels_for_structure_stack:
+                self.labels_for_structure_stack[-1].update(labels_for_structure)
+            else:
+                self.tags.update([structural_coverage(l) for l in labels_for_structure])
```

Comment:
`set.update` can take an iterable, so we don't need the `[]` to turn it into a list first :smile: 

(I guess `flake8-comprehensions` doesn't have interproceedural type information...)

Reply:
We don't, but generally list comprehensions are faster than generator comprehensions if you're going to consume the whole thing. I doubt the performance makes any difference but I tend to err on the side of using list comprehensions.